aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterClearTest.java

```diff
@@ -0,0 +1,210 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.reactivex.disposables.Disposable;
+import io.reactivex.observers.TestObserver;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public final class SQLiteStorageAdapterClearTest {
+    private static final String DATABASE_NAME = "AmplifyDatastore.db";
+
+    private SynchronousStorageAdapter adapter;
+    private Context context;
+    private TestObserver<StorageItemChange<? extends Model>> observer;
+    private AtomicReference<Disposable> subscriberDisposableRef = new AtomicReference<>();
+    private TestFileObserver fileObserver;
+
+    @BeforeClass
+    public static void enableStrictMode() {
+        StrictMode.enable();
+    }
+
+    @Before
+    public void setup() {
+        TestStorageAdapter.cleanup();
+        context = ApplicationProvider.getApplicationContext();
+        adapter = TestStorageAdapter.create(AmplifyModelProvider.getInstance());
+        //Set subscriberDisposableRef = <value received from RxJava>.
+        //Needed so we can make assertions on the state of the subscriber later.
+        observer = adapter
+            .observe()
+            .doOnSubscribe(subscriberDisposableRef::set)
+            .test();
+
+        fileObserver = new TestFileObserver(Objects.requireNonNull(context.getDatabasePath(DATABASE_NAME).getParent()));
+        fileObserver.startWatching();
+    }
+
+    @After
+    public void teardown() {
+        fileObserver.stopWatching();
+        TestStorageAdapter.cleanup(adapter);
+    }
+
+    /**
+     * Save a record to the database and verify it was saved.
+     * Then call clear and verify that the database file is re-created
+     * and is writable.
+     * @throws DataStoreException bubbles up exceptions thrown from the adapter
+     * @throws IOException if it can't ready file creation time
+     * @throws InterruptedException if something happens when sleeping for 1 second
+     */
+    @Test
+    public void clearDeletesAndRecreatesDatabase() throws DataStoreException, IOException, InterruptedException {
+        assertDbFileExists();
+        assertEquals(0, fileObserver.createFileEventCount);
+        assertEquals(0, fileObserver.deleteFileEventCount);
+        BlogOwner blogger1 = createBlogger("Dummy Blogger Sr.");
+        BlogOwner blogger2 = createBlogger("Dummy Blogger Jr.");
+        //Save a record and check if it's there
+        adapter.save(blogger1);
+        assertRecordIsInDb(blogger1.getName());
+        //Verify observer is still alive
+        assertFalse(subscriberDisposableRef.get().isDisposed());
+        assertObserverReceivedRecord(blogger1);
+        //The ensures files are created at least 1 second apart
+        Thread.sleep(1000);
```

Comment:
But how come? Is it necessary?

Reply:
It's not needed anymore. Before I started using the `FileObserver`, I was checking the created date of the file. Had to move away from that because creation date attribute is not available in older versions of Android.